lib/zip_source_buffer.c

```diff
@@ -159,11 +160,7 @@ zip_source_buffer_fragment_with_attributes_create(const zip_buffer_fragment_t *f
     ctx->out = NULL;
     ctx->mtime = time(NULL);
     if (attributes) {
-#if __STDC_WANT_SECURE_LIB__
-        memcpy_s(&ctx->attributes, sizeof(ctx->attributes), attributes, sizeof(ctx->attributes));
-#else
-        memcpy(&ctx->attributes, attributes, sizeof(ctx->attributes));
-#endif
+        _zip_memcpy(&ctx->attributes, sizeof(ctx->attributes), attributes, sizeof(ctx->attributes));
```

Comment:
Should that be `_zip_memcpy(&ctx->attributes, sizeof(ctx->attributes), attributes, sizeof(*attributes));`?